nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -458,6 +471,13 @@ public void process(final InputStream rawIn) throws IOException {
                             // single part upload
                             //----------------------------------------
                             final PutObjectRequest request = new PutObjectRequest(bucket, key, in, objectMetadata);
+                            if (keyId != null) {
+                                    if (!context.getProperty(SIGNER_OVERRIDE).getValue().equals("AWSS3V4Signer")) {
+                                            getLogger().error("Uploading with AWS:KMS requires S3V4 signature, please enable it");
+                                            return;
```

Comment:
The `return` here routes the flowfile to success without going to S3.  Is that intended?